scripts/imgtool/image.py

```diff
@@ -129,7 +138,7 @@ def check(self):
                 raise Exception(msg)
 
     def sign(self, key):
-        self.add_header(key)
+        self.add_header(key, self.endian)
```

Comment:
There's no need to pass `self.endian` here since it's already data from the object.